### ISSUE TITLE
[4.0] crash when constructing a Vector3 with the delta parameter from _process(delta)

### ISSUE THREAD

**Issue by Arakash** [godotengine/godot#42024]
**Godot version:**
Godot Engine v4.0.dev.custom_build.43725691b

**OS/device including version:**
Windows 10, Vulkan

**Issue description:**
When constructing a Vector3 inside the _process function with the delta parameter, the game crashes.

**Steps to reproduce:**
```gdscript
func _process(delta):
	var movement = Vector3(0, 0, delta)
```
I also tried explicitly making the zeros into floats, but there is still a crash:
```gdscript
func _process(delta):
	var movement = Vector3(0.0, 0.0,delta)
```

**Minimal reproduction project:**
[test.zip](https://github.com/godotengine/godot/files/5213942/test.zip)

**Comment by Calinou** [comment]
Backtrace:

```text
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib64/libc.so.6(+0x3c6b0) [0x7f0e32caf6b0] (??:0)
[2] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/hugo/Documents/Git/godotengine/godot/modules/gdscript/gdscript_function.cpp:953)
[3] GDScriptInstance::call(StringName const&, Variant const**, int, Callable::CallError&) (/home/hugo/Documents/Git/godotengine/godot/modules/gdscript/gdscript.cpp:1317)
[4] ScriptInstance::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/core/script_language.cpp:?)
[5] Node::_notification(int) (/home/hugo/Documents/Git/godotengine/godot/scene/main/node.cpp:58)
[6] Node::_notificationv(int, bool) (/home/hugo/Documents/Git/godotengine/godot/./scene/main/node.h:46)
[7] CanvasItem::_notificationv(int, bool) (/home/hugo/Documents/Git/godotengine/godot/./scene/main/canvas_item.h:?)
[8] Control::_notificationv(int, bool) (/home/hugo/Documents/Git/godotengine/godot/./scene/gui/control.h:?)
[9] Object::notification(int, bool) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:806)
[10] SceneTree::_notify_group_pause(StringName const&, int) (/home/hugo/Documents/Git/godotengine/godot/scene/main/scene_tree.cpp:818)
[11] SceneTree::idle(float) (/home/hugo/Documents/Git/godotengine/godot/scene/main/scene_tree.cpp:447)
[12] Main::iteration() (/home/hugo/Documents/Git/godotengine/godot/main/main.cpp:2422)
[13] OS_LinuxBSD::run() (/home/hugo/Documents/Git/godotengine/godot/platform/linuxbsd/os_linuxbsd.cpp:240)
[14] /home/hugo/Documents/Git/godotengine/godot/bin/godot.linuxbsd.tools.64.llvm(main+0x1dc) [0x3f8a2cc] (/home/hugo/Documents/Git/godotengine/godot/platform/linuxbsd/godot_linuxbsd.cpp:58)
[15] /lib64/libc.so.6(__libc_start_main+0xf3) [0x7f0e32c9a1a3] (??:0)
[16] /home/hugo/Documents/Git/godotengine/godot/bin/godot.linuxbsd.tools.64.llvm(_start+0x2e) [0x3f8a02e] (??:?)
-- END OF BACKTRACE --
```

**Comment by Arakash** [comment]
@Calinou  thanks!

**Comment by akien-mga** [comment]
Fixed by #41881.
